manifests/charts/istio-control/istio-discovery/templates/mutatingwebhook.yaml

```diff
@@ -22,8 +22,102 @@ a unique prefix to each. */}}
   failurePolicy: Fail
   admissionReviewVersions: ["v1beta1", "v1"]
 {{- end }}
-{{- /* Installed for each revision - not installed for cluster resources ( cluster roles, bindings, crds) */}}
+{{- /* Core-default defines the common configuration used by the default webhook segments */}}
+{{- define "core-default" }}
+- name: {{.Prefix}}sidecar-injector.istio.io
+  clientConfig:
+    {{- if .Values.istiodRemote.injectionURL }}
+    url: {{ .Values.istiodRemote.injectionURL }}
+    {{- else }}
+    service:
+      name: istiod
+      namespace: {{ .Release.Namespace }}
+      path: "/inject"
+    {{- end }}
+    caBundle: ""
+  sideEffects: None
+  rules:
+    - operations: [ "CREATE" ]
+      apiGroups: [""]
+      apiVersions: ["v1"]
+      resources: ["pods"]
+  failurePolicy: Fail
+  admissionReviewVersions: ["v1beta1", "v1"]
+{{- end }}
 {{- if not .Values.global.operatorManageWebhooks }}
+{{- /* Default mutating webhook - single instance should exist across all revisions
+Needs to be conditionally created so that it is marked with the correct revision label. This allows
+for the webhook to be patched by the default revision and remain intact when uninstalling other revisions */}}
+{{- if or (eq .Values.revision "") (eq .Values.defaultRevision true) }}
+{{- if not .Values.sidecarInjectorWebhook.useLegacySelectors}}
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: MutatingWebhookConfiguration
+metadata:
+{{- if eq .Release.Namespace "istio-system"}}
+  name: istio-sidecar-injector-default
```

Comment:
So when doing `istioctl install`, the default injection logic will be in `istio-sidecar-injector-default`, and it will get overwritten when doing `istioctl install --set revision=foo --defaultRevision`.

This is somewhat confusing since the default install will result in two `MutatingWebhookConfiguration`s, `istio-sidecar-injector` with no webhooks (since all webhooks in that config are per-revision), and `istio-sidecar-injector-default` with 2

Reply:
Oh, I see. I was expecting the "default" one to be `istio-sidecar-injector` and the per revision one to be `istio-sidecar-injector-<revision>`. This seems a bit confusing to me and I am not sure it actually works.
```
$ ~/istio/istio-1.9.0/bin/istioctl install -y
$ grun ./istioctl/cmd/istioctl install -y -d manifests --set revision=bar --set defaultRevision=true
$ kgmwh -L istio.io/rev -L istio.io/default
NAME                             WEBHOOKS   AGE    REV       DEFAULT
istio-sidecar-injector           1          112s   default
istio-sidecar-injector-bar       2          52s    bar
istio-sidecar-injector-default   2          100s   bar       true
```

both istio-sidecar-injector and istio-sidecar-injector-default are matching istio-injection=enabled

Reply:
Ah I see now, you're right, we definitely need to overwrite the `istio-sidecar-injector` webhook from the old default install.

Reply:
Looks like `istioctl install` actually tries to merge resources with the same name rather than overwriting, which breaks the logic here for both overwriting the `istiod` service and the `istio-sidecar-injector` webhook (for example: adds labels to service selector rather than overwriting them). Trying to figure out how to get around this

Reply:
I have seen a number of issues about this behavior. @richardwxn @ostromart any ideas?

Reply:
Is there a way to explicitly patch it with a K8sObjectOverlay? https://github.com/istio/istio/blob/d706ab80df4b1742b299b6c38bf80cb967697efc/pkg/config/labels/instance.go#L97

Reply:
Even after using the `K8sObjectOverlay` to change the generated config I'm not sure the installer wouldn't just go forward with a merge rather than an overwrite when actually applying it (haven't tried this out though, could be wrong)

Ideally there'd be an annotation we could put on the resource to let the installer know it should overwrite rather than attempt a merge?